shared/pkg/config/runtime_scanner.go

```diff
@@ -40,19 +49,131 @@ func setRegistryConfigDefaults() {
 func LoadRuntimeScannerRegistryConfig(imageID string) *Registry {
 	setRegistryConfigDefaults()
 
-	username, password, err := creds.ExtractCredentials(imageID)
-	if err != nil {
-		log.Warnf("Failed to extract credentials for imageID %v: %v", imageID, err)
+	var auth Auth
+
+	imagePullSecretPath := viper.GetString(ImagePullSecretPath)
+	if imagePullSecretPath != "" {
+		authConfig, err := GetAuthConfigFromPullSecrets(imagePullSecretPath, imageID)
+		if err != nil {
+			log.Fatalf("Failed to resolve auth config from pull secrets: %v", err)
```

Comment:
Before we just printed a warning and continued, i think we should do the same here. At worst we will fail to scan due to missing credentials, at best we will succeed :)